ui/apps/platform/src/Containers/Integrations/IntegrationTilesPage/IntegrationTilesPage.tsx

```diff
@@ -1,70 +1,104 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
+import React, { ReactElement } from 'react';
+import { useSelector } from 'react-redux';
 import { createStructuredSelector } from 'reselect';
 import { PageSection, Title } from '@patternfly/react-core';
 
 import useCentralCapabilities from 'hooks/useCentralCapabilities';
 import useFeatureFlags from 'hooks/useFeatureFlags';
 import { integrationsPath } from 'routePaths';
 import { selectors } from 'reducers';
-import integrationsList from '../utils/integrationsList';
+import { ClusterInitBundle } from 'services/ClustersService';
+import { ApiToken } from 'types/apiToken.proto';
+import { BackupIntegration } from 'types/externalBackup.proto';
+import { ImageIntegration } from 'types/imageIntegration.proto';
+import { NotifierIntegration } from 'types/notifier.proto';
+import { SignatureIntegration } from 'types/signatureIntegration.proto';
+
+import integrationsList, { IntegrationDescriptor } from '../utils/integrationsList';
 
 import IntegrationTile from './IntegrationTile';
 import IntegrationsSection from './IntegrationsSection';
 
-const IntegrationTilesPage = ({
-    apiTokens,
-    clusterInitBundles,
-    backups,
-    imageIntegrations,
-    notifiers,
-    signatureIntegrations,
-}) => {
+type IntegrationSource = keyof typeof integrationsList;
```

Comment:
Since this is already implicitly typed by the shape of `IntegrationDescriptorMap` in `integrationsList.ts#90-96, would it be better to define it in that file and just import it here?